streams/src/test/java/org/apache/kafka/streams/integration/KTableKTableForeignKeyJoinIntegrationTest.java

```diff
@@ -27,673 +25,314 @@
-        //However, it will still run the join function since this is LEFT join.
-        List<KeyValue<Integer, Float>> leftTableEvents = Arrays.asList(new KeyValue<>(1, 8.33f));
-        final List<KeyValue<String, Long>> rightTableEvents = Arrays.asList(new KeyValue<>("1", 10L)); //partition 0
-        IntegrationTestUtils.produceKeyValuesSynchronously(LEFT_TABLE, leftTableEvents, LEFT_PROD_CONF, MOCK_TIME);
-        IntegrationTestUtils.produceKeyValuesSynchronously(RIGHT_TABLE, rightTableEvents, RIGHT_PROD_CONF, MOCK_TIME);
-
-        final String currentMethodName = new Object() { }
-                .getClass()
-                .getEnclosingMethod()
-                .getName();
-        createAndStartStreamsApplication(currentMethodName, true);
-
-        final List<KeyValue<Integer, String>> expected = new LinkedList<>();
-        expected.add(new KeyValue<>(1, "value1=8.33,value2=null"));
-        final List<KeyValue<Integer, String>> result = IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(
-                CONSUMER_CONFIG,
-                OUTPUT,
-                expected.size());
-        assertEquals(expected, result);
-
-        //There is also no matching extracted foreign-key for 18 anywhere.
-        //However, it will still run the join function since this if LEFT join.
-        leftTableEvents = Arrays.asList(new KeyValue<>(1, 18.0f));
-        IntegrationTestUtils.produceKeyValuesSynchronously(LEFT_TABLE, leftTableEvents, LEFT_PROD_CONF, MOCK_TIME);
-
-        final List<KeyValue<Integer, String>> expected2 = new LinkedList<>();
-        expected2.add(new KeyValue<>(1, "value1=18.0,value2=null"));
-        final List<KeyValue<Integer, String>> result2 = IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(
-                CONSUMER_CONFIG,
-                OUTPUT,
-                expected2.size());
-        assertEquals(expected2, result2);
-
-
-        leftTableEvents = Arrays.asList(new KeyValue<>(1, 1.11f));
-        IntegrationTestUtils.produceKeyValuesSynchronously(LEFT_TABLE, leftTableEvents, LEFT_PROD_CONF, MOCK_TIME);
-
-        final List<KeyValue<Integer, String>> expected3 = new LinkedList<>();
-        expected3.add(new KeyValue<>(1, "value1=1.11,value2=10"));
-        final List<KeyValue<Integer, String>> result3 = IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(
-                CONSUMER_CONFIG,
-                OUTPUT,
-                expected3.size());
-        assertEquals(expected3, result3);
-
-        //Ensure the state stores have the correct values within:
-        final Set<KeyValue<Integer, String>> expMatResults = new HashSet<>();
-        expMatResults.add(new KeyValue<>(1, "value1=1.11,value2=10"));
-        validateQueryableStoresContainExpectedKeyValues(expMatResults, currentMethodName);
+    public void doJoinFromLeftThenDeleteLeftEntity() {
+        for (final Boolean leftJoin : new Boolean[] {false, true}) {
```

Comment:
Reusing the code is fine, but I would not do it in a loop, but with two proper test methods (with corresponding names) and a single private method that is called by both passing int `leftJoin` as parameter.

Reply:
Sounds good. I've made this a parameterized test, which amounts to the same thing. Not sure what I was thinking...